### PR TITLE
udiskslinuxblock.c: Use common device handling func.

### PR DESCRIPTION
**Pull request by tasleson**
I experimented with using a common authentication function, but in the end the changes to support that outweigh code saving and complexity savings.  Some functions are calling into authenticate multiple times.  Furthermore trying to make a common authentication function across the code base is hampered in the different ways it occurs and the amount of associated data would need to be returned as well.

### CODE REVIEW

**Comment by vojtechtrefny** [comment]
Jenkins, ok to test.

File: src/udiskslinuxblock.c [hunk 1]
```diff
@@ -434,6 +434,34 @@ update_hints (UDisksLinuxBlock  *block,
...
+        {
+          for (n = 0; symlinks[n] != NULL; n++)
+            {
+              if (g_strcmp0 (device, symlinks[n]) == 0)
+                {
+                  *ret = g_list_prepend (*ret, g_object_ref (object));
+                }
+            }
+        }
+    }
```
Comment (vpodzime) [inline src/udiskslinuxblock.c]: I myself tend to use curly brackets in C even in places where they are not needed. But due to the coding style used in udisks, this is really crazy. I think getting rid of all those extra curly brackets here would make the function much better readable.

**Comment by tasleson** [comment]
@vpodzime Removed extra "{}", thanks for the review.

**Comment by vpodzime** [comment]
> Removed extra "{}", thanks for the review.

I think even more could go away. At least the ones in the ``for`` cycle.

**Comment by tasleson** [comment]
@vpodzime Wish granted :-)

**Comment by vpodzime** [comment]
> @vpodzime Wish granted :-)

Cool, thanks!

**Comment by vpodzime** [comment]
Jenkins, test this, please.

**Pull request merged**